x/cronos/keeper/keeper.go

```diff
@@ -217,17 +217,10 @@ func (k Keeper) GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.Accou
 // RegisterOrUpdateTokenMapping update the token mapping, register a coin metadata if needed
 func (k Keeper) RegisterOrUpdateTokenMapping(ctx sdk.Context, msg *types.MsgUpdateTokenMapping) error {
 	if types.IsSourceCoin(msg.Denom) {
-		contract, err := types.GetContractAddressFromDenom(msg.Denom)
+		_, err := types.GetContractAddressFromDenom(msg.Denom)
 		if err != nil {
 			return err
 		}
-		// we check that denom use the same contract address in checksum format
-		if contract != common.HexToAddress(msg.Contract).Hex() {
```

Comment:
removed this check otherwise we cannot change the token mapping of source token (cronos0x...) to another contract address